picoquicfirst/picoquicdemo.c

```diff
@@ -760,33 +766,35 @@ static void first_client_callback(picoquic_cnx_t* cnx,
                 (int)stream_ctx->received_length);
         }
         return;
-    } else if (fin_or_event == picoquic_callback_stop_sending) {
+    }
+
+    if (fin_or_event == picoquic_callback_stop_sending) {
         char buf[256];
         picoquic_reset_stream(cnx, stream_id, 0);
 
         fprintf(stdout, "Stop sending received on stream %d, command: %s\n",
             stream_ctx->stream_id,
             strip_endofline(buf, sizeof(buf), (char*)&stream_ctx->command));
         return;
-    } else {
-        if (length > 0) {
-            (void)fwrite(bytes, 1, length, stream_ctx->F);
-            stream_ctx->received_length += length;
-        }
+    }
+    
+    if (length > 0) {
+        (void)fwrite(bytes, 1, length, stream_ctx->F);
+        stream_ctx->received_length += length;
+    }
 
-        /* if FIN present, process request through http 0.9 */
-        if (fin_or_event == picoquic_callback_stream_fin) {
-            char buf[256];
-            /* if data generated, just send it. Otherwise, just FIN the stream. */
-            fclose(stream_ctx->F);
-            stream_ctx->F = NULL;
-            ctx->nb_open_streams--;
-            fin_stream_id = stream_id;
+    /* if FIN present, process request through http 0.9 */
+    if (fin_or_event == picoquic_callback_stream_fin) {
+        char buf[256];
+        /* if data generated, just send it. Otherwise, just FIN the stream. */
+        fclose(stream_ctx->F);
+        stream_ctx->F = NULL;
+        ctx->nb_open_streams--;
+        fin_stream_id = stream_id;
 
-            fprintf(stdout, "Received file %s, after %d bytes, closing stream %d\n",
+        fprintf(stdout, "Received file %s, after %d bytes, closing stream %d\n",
                 strip_endofline(buf, sizeof(buf), (char*)&stream_ctx->command[4]),
                 (int)stream_ctx->received_length, stream_ctx->stream_id);
-        }
     }
 
```

Comment:
Again, these are probably good fixes, but unrelated. Should be moved to a separate PR.